### PR TITLE
Updated Button docs

### PR DESCRIPTION
**Pull request by ponori**
### Summary
Unified documentation of the Button component.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "packages/*"
   ],
   "dependencies": {
+    "gestalt-datepicker": "^33.4.3",
```
Comment (christianvuerings) [inline package.json:8]: @ponori could we 
* revert the changes to `package.json` and `yarn.lock`?
* remove the `package-lock.json` file? We use `yarn` so that should not show up

**Comment by somerandomdude** [comment]
Ugh, sorry for that. I'll make that fix. Thanks!

File: package.json (hunk 1 above)
Comment (somerandomdude) [reply package.json:8]: Ugh, sorry for that. I'll make that fix. Thanks!

**Comment by ponori** [comment]
@ayeshakmaz - Any thoughts on how to change two props in the `CombinationNew` component? I'd like to provide custom text for each color button in https://deploy-preview-1654--gestalt.netlify.app/button#Color-on-white-backgrounds.

File: docs/pages/button.js [hunk 2]
```diff
@@ -27,88 +32,75 @@ card(
         required: false,
         defaultValue: null,
         description: [
-          'Supply a short, descriptive label for screen-readers to replace button texts that do not provide sufficient context about the button component behavior. Texts like `Click Here,` `Follow,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the button text.',
-          'Use IconButton if you only need a button with an icon and no text.',
-          'Accessibility: It populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of `text`.',
+          'Label to provide more context around Button’s function or purpose. See the [Accessibility guidelines](/accessibility) to learn more.',
         ],
-        href: 'accessibilityLabel',
+        href: 'Accessibility-props:-controls-expanded-and-popup',
```
Comment (AlbertCarreras) [inline docs/pages/button.js:37]: the new docs don't support links on the prop names. You can remove all this href sections.

File: docs/pages/button.js [hunk 3]
```diff
@@ -127,58 +119,48 @@ card(
...
       },
       {
         name: 'text',
         type: 'string',
         required: true,
         defaultValue: null,
         description: [
-          'Text to render inside the button to convey the function and purpose of the button. The button text has a fixed size.',
+          'Text to render inside the Button to convey the function and purpose of the Button. The Button text has a fixed size.',
           'Accessibility: Screen readers read the `accessibilityLabel` prop, if present, instead of the `text` prop. See `accessibilityLabel` for more info.',
```
Comment (AlbertCarreras) [inline docs/pages/button.js:132]: I would move this portion 

"Accessibility: Screen readers read the `accessibilityLabel` prop, if present, instead of the `text` prop. See `accessibilityLabel` for more info.'" to the a11y section.

File: docs/pages/button.js [hunk 4]
```diff
@@ -27,88 +32,75 @@ card(
...
       {
         name: 'accessibilityExpanded',
         type: 'boolean',
         required: false,
         defaultValue: null,
         description: [
-          'Indicate that a button component hides or exposes collapsible components and expose whether they are currently expanded or collapsed.',
-          'Optional with button-role + button-type buttons.',
-          'Accessibility: It populates aria-expanded.',
+          'Needed if Button controls a grouping of other elements, e.g. Dropdown or Flyout. This is used to indicate if the controlled grouping is currently expanded or collapsed. See the [Accessibility guidelines](/accessibility) to learn more.',
```
Comment (AlbertCarreras) [inline docs/pages/button.js:55]: I think `Button controls a grouping` can be simplified to 
"Button controls the visibility of other elements, e.g. Dropdown or Flyout"

I think `grouping` is unrelated

File: docs/pages/button.js [hunk 5]
```diff
@@ -187,46 +169,34 @@ card(
...
+        href: 'roles',
       },
       {
         name: 'rel',
         type: `'none' | 'nofollow'`,
         required: false,
         defaultValue: 'none',
-        description: 'Optional with link-role buttons.',
-        href: 'type-roles',
+        description: 'Optional with link-role Buttons.',
```
Comment (AlbertCarreras) [inline docs/pages/button.js:188]: here we should link to variant rel & target.

I would follow what we did for IconButton. https://deploy-preview-1654--gestalt.netlify.app/iconbutton#role_linkProps

Comment (AlbertCarreras) [reply docs/pages/button.js:188]: link-role needs `link-role`

Comment (AlbertCarreras) [reply docs/pages/button.js:188]: also the color of the text for whatever reason is different!

File: docs/pages/button.js [hunk 6]
```diff
@@ -187,46 +169,34 @@ card(
...
         required: false,
         defaultValue: 'button',
-        description: [
-          `Select a button variant:`,
-          `- 'button': Use to render 'submit' or 'button'-type buttons. The button is rendered as a \`<button>\`.`,
-          `- 'link': Use for buttons to act like links. The button is rendered as an \`<a>\`.`,
-          `Required with link-role buttons.`,
-        ],
-        href: 'type-roles',
+        description: ['Use “link” to indicate Button that is acting as an `<a>` link.'],
```
Comment (AlbertCarreras) [inline docs/pages/button.js:180]: i'd create a variant for roles. If OOO, I'd open a JIra ticket to mirror IconButton

File: docs/pages/button.js [hunk 7]
```diff
@@ -259,245 +229,481 @@ card(
...
+        type="do"
+        description="Keep the Button text as simple and actionable as possible. Refer to the [Button writing guidelines](#Writing) for more detail."
+        defaultCode={`
+  <Button text="Create account" size="lg" color="red" />
+`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        description="Do not add icons to a Button to refinforce the text."
```
Comment (AlbertCarreras) [inline docs/pages/button.js:301]: typo reinforce

File: docs/pages/button.js [hunk 8]
```diff
@@ -259,245 +229,481 @@ card(
...
+<Flex gap={2}>
+  <Button text="Kontoeinst..." size="lg" color="gray" />
+  <Button text="Neues We..." size="lg" color="red" />
+</Flex>
+`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="do"
+        description="Keep the Button text as simple and actionable as possible. Refer to the [Button writing guidelines](#Writing) for more detail."
```
Comment (AlbertCarreras) [inline docs/pages/button.js:293]: I'd also refer to a11y guidelines, if text is not enough for a11y and needs label.

File: docs/pages/button.js [hunk 9]
```diff
@@ -259,245 +229,481 @@ card(
...
-  <Button size="sm" text="Small-sized button" />
-  <Button text="Medium-sized button" />
-  <Button size="lg" text="Large-sized button" />
-</Flex>`}
-  />,
+  <MainSection name="Accessibility">
+    <MainSection.Subsection
+      title="ARIA attributes"
+      description={`
+When Button text does not provide sufficient context about the Button’s behavior, supply a short, descriptive label for screen-readers using accessibilityLabel. Texts like “Click here“, “Follow“, or “Shop“ can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text with deeper context to replace the Button text, like “Follow Ryan” or “Shop Wedding Invitations”.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:315]: using `accessibilityLabel`  missing format

Comment (AlbertCarreras) [reply docs/pages/button.js:315]: Texts like “Click...

 should go in new line

Comment (AlbertCarreras) [reply docs/pages/button.js:315]: If Button ... 

needs one more line of space

File: docs/pages/button.js [hunk 10]
```diff
@@ -259,245 +229,481 @@ card(
...
-  <Combination
-    id="color"
-    name="Color"
-    color={['gray', 'red', 'blue', 'white', 'transparent', 'semiTransparentWhite']}
-  >
-    {(props, i) => <Button id={`example-${i}`} onChange={() => {}} {...props} text="Button" />}
-  </Combination>,
+  <MainSection
+    name="Localization"
+    description="Be sure to localize `text` and `accessibilityLabel`. Note that localization can lengthen text by 20 to 30 percent. Avoid truncating Button text whenever possible."
```
Comment (AlbertCarreras) [inline docs/pages/button.js:328]: I'd refer back to best practices for truncation

File: docs/pages/button.js [hunk 11]
```diff
@@ -127,58 +119,48 @@ card(
...
+        href: 'Size',
       },
       {
         name: 'text',
         type: 'string',
         required: true,
         defaultValue: null,
         description: [
-          'Text to render inside the button to convey the function and purpose of the button. The button text has a fixed size.',
+          'Text to render inside the Button to convey the function and purpose of the Button. The Button text has a fixed size.',
```
Comment (AlbertCarreras) [inline docs/pages/button.js:131]: I'd move `The Button text has a fixed size` to the variant section (Size variant) We explain

Button is available in 3 fixed sizes: [...] The Button text has always a fixed size of xx pixels.

File: docs/pages/button.js [hunk 12]
```diff
@@ -259,245 +229,481 @@ card(
...
+      <CombinationNew size={['sm', 'md', 'lg']}>
+        {({ size }) => (
+          <Button accessibilityLabel={`Example size ${size}`} color="red" text="Save" size={size} />
+        )}
+      </CombinationNew>
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Width"
+      description={`
+1. Inline (default)
```
Comment (AlbertCarreras) [inline docs/pages/button.js:353]: default or `fullWidth={false}`

File: docs/pages/button.js [hunk 13]
```diff
@@ -259,245 +229,481 @@ card(
...
-          </Box>
-          <Box padding={2} position="absolute" bottom right>
+        )}
+      </CombinationNew>
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Color on white backgrounds"
+      description={`
+1. Red (Primary)
+    High emphasis, used for primary actions.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:374]: can we give some hints of what's a primary/secondary/tertiary action?

File: docs/pages/button.js [hunk 14]
```diff
@@ -259,245 +229,481 @@ card(
...
-            <Button
-            color={selectedSemiTransparentWhite ? 'white' : 'semiTransparentWhite'}
-            iconEnd="pin-hide"
-            text="Hide this image"
-            inline />
-          </Box>
-        </Image>
       </Box>
-    </Flex>
+      {open &&
```
Comment (AlbertCarreras) [inline docs/pages/button.js:430]: image
 It took me a while to realize it wasn't a white button but a Tab. Maybe the anchor could be something else.

File: docs/pages/button.js [hunk 15]
```diff
@@ -259,245 +229,481 @@ card(
...
-    id="selected"
-    defaultCode={`
-function Example() {
-  const [selected, setSelected] = React.useState(false);
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Icons"
+      description={`
+1. Icon end
```
Comment (AlbertCarreras) [inline docs/pages/button.js:526]: we don't need the one here.

File: docs/pages/button.js [hunk 16]
```diff
@@ -259,245 +229,481 @@ card(
...
+Adds an icon after the Button text. Icons should only be used to reinforce a specific function or interaction of the Button. Menus and external links are a common use case. Use \`arrow-up-right\` when linking to an external URL or \`arrow-down\` when displaying a Popover on click. Note that iconEnd on Button is not accessible to screen readers.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:527]: to reinforce

to _visually?_ reinforce

Comment (AlbertCarreras) [reply docs/pages/button.js:527]: `Note that iconEnd on Button is not accessible to screen readers. 
`
Not sure if this is needed.

File: docs/pages/button.js [hunk 17]
```diff
@@ -259,245 +229,481 @@ card(
...
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      columns={2}
+      title="Role"
+      description={`
+1. Button (default)
+    The “button” \`role\` is used for actions. This is the default and should be used for most Buttons.
+2. Link
+    The “link” \`role\` is used for navigating by URL. These Buttons should not use a \`selected\` state.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:549]: VOILA!!! Check this conversation https://pinterest.slack.com/archives/G01GN7CE99S/p1631231159047700

File: docs/pages/button.js [hunk 18]
```diff
@@ -259,245 +229,481 @@ card(
...
+  role="link"
+  href="https://pinterest.com"
+/>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="rel and target"
+      description={`
+These optional props control the behavior of \`role="link"\` Buttons. External links commonly use \`target="_blank"\` to open the link in a new tab or window, and \`rel="nofollow"\` to provide hints for SEO.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:580]: image


Could we fix the typos in `IconButton`?

rel is optional. Use "**nofollow**" for external links to specify to web crawlers not **to** follow the link.

File: docs/pages/button.js [hunk 19]
```diff
@@ -259,245 +229,481 @@ card(
...
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      columns={2}
+      title="States"
+      description={`
+1. Default
+    The typical state of a Button that represents it can be interacted with and is not in a selected state.
+2. Disabled
+Used to block user interaction such as hover, focus and click. Disabled Buttons are completely unreachable by a keyboard and screenreader, so do not attach Tooltips to disabled Buttons. If you need to explain why a Button is disabled, use an IconButton + Tooltip next to the disabled Button.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:606]: I feel this could be a Do/Don't


"If you need to explain why a Button is disabled, use an IconButton + Tooltip next to the disabled Button."

I don't understand/visualize this.

File: docs/pages/button.js [hunk 20]
```diff
@@ -259,245 +229,481 @@ card(
...
+3. Selected
+  When Button is used to toggle a boolean state or control the visibility of other elements (e.g. Dropdown), use the \`selected\` prop to indicate the current state. Do not use this prop with \`role="link"\` Buttons.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:608]: Should we include Tabs functionality? as per conversation here: 
https://pinterest.slack.com/archives/G01GN7CE99S/p1631231159047700

File: docs/pages/button.js [hunk 21]
```diff
@@ -513,89 +719,52 @@ function ButtonPopoverExample() {
...
-  );
-}
+  <MainSection name="Writing">
+    <MainSection.Subsection columns={2}>
+      <MainSection.Card
+        cardSize="md"
+        type="do"
+        description={`
+- If your object is already described on the screen, Buttons only need a verb.
+- If your object isn’t described on the screen, Buttons need a verb + the object.
```
Comment (AlbertCarreras) [inline docs/pages/button.js:737]: can we add an example?

File: docs/pages/button.js [hunk 22]
```diff
@@ -513,89 +719,52 @@ function ButtonPopoverExample() {
...
+      />
+    </MainSection.Subsection>
+  </MainSection>,
 );
 
 card(
   <MainSection name="Related">
     <MainSection.Subsection
       description={`
+**[ButtonGroup](/buttongroup)**
```
Comment (AlbertCarreras) [inline docs/pages/button.js:757]: we need some spacing here

**Review by AlbertCarreras — APPROVED** [review]
Overall looks good. We still need to do the Eng guidelines, so we can give a second pass then.

File: docs/pages/button.js (hunk 15 above)
Comment (ponori) [reply docs/pages/button.js:526]: Good call. Done!

File: docs/pages/button.js (hunk 2 above)
Comment (ponori) [reply docs/pages/button.js]: Thanks! Done!

File: docs/pages/button.js (hunk 4 above)
Comment (ponori) [reply docs/pages/button.js:53]: Good call. Fixed!

File: docs/pages/button.js (hunk 3 above)
Comment (ponori) [reply docs/pages/button.js:122]: Done!

File: docs/pages/button.js (hunk 7 above)
Comment (ponori) [reply docs/pages/button.js:285]: Nice catch! Done!

File: docs/pages/button.js (hunk 8 above)
Comment (ponori) [reply docs/pages/button.js:277]: Done!

File: docs/pages/button.js (hunk 9 above)
Comment (ponori) [reply docs/pages/button.js:299]: Done!

File: docs/pages/button.js (hunk 11 above)
Comment (ponori) [reply docs/pages/button.js:121]: Done!

File: docs/pages/button.js (hunk 17 above)
Comment (ponori) [reply docs/pages/button.js:533]: I can't see the conversation. :(

File: docs/pages/button.js (hunk 21 above)
Comment (ponori) [reply docs/pages/button.js:721]: Done!

File: docs/pages/button.js (hunk 22 above)
Comment (ponori) [reply docs/pages/button.js:741]: Done!

File: docs/pages/button.js (hunk 19 above)
Comment (ponori) [reply docs/pages/button.js:590]: Done!

File: docs/pages/button.js (hunk 18 above)
Comment (ponori) [reply docs/pages/button.js:564]: Not sure if I follow - is this a change you'd like to make to IconButton?

File: docs/pages/button.js (hunk 16 above)
Comment (ponori) [reply docs/pages/button.js:511]: Done! If I recall, Ayesha suggested to add the bit about iconEnd not being accessible to screen readers

File: docs/pages/button.js (hunk 14 above)
Comment (ponori) [reply docs/pages/button.js:414]: Me thinks we can backlog this as an enhancement - especially given this example was based on one from Popover, so we'd likely want to fix both at once.

File: docs/pages/button.js (hunk 13 above)
Comment (ponori) [reply docs/pages/button.js:358]: I think this is a great thing to add down the road because, to be honest, I think we need to do more thinking on this topic before we can provide a cogent answer..

File: docs/pages/button.js (hunk 10 above)
Comment (ponori) [reply docs/pages/button.js:312]: Done!

File: docs/pages/button.js (hunk 6 above)
Comment (ponori) [reply docs/pages/button.js:167]: Creating a ticket!

File: docs/pages/button.js (hunk 5 above)
Comment (ponori) [reply docs/pages/button.js:174]: This also seems like it needs some additional work. Ticketing!

File: docs/pages/button.js (hunk 20 above)
Comment (ponori) [reply docs/pages/button.js:592]: I would suggest against this. The selected state is specifically for situations where you'r toggling states of a specific button (Follow/Following) (Save/Saved)

Comment (ponori) [reply docs/pages/button.js:592]: OK, I added a Don't for this usecase, but given there's a style bug at the moment, I think we should probably fix this before publishing the new Button guidelines. Thoughts?

File: docs/pages/button.js (hunk 16 above)
Comment (ayeshakmaz) [reply docs/pages/button.js]: I think it's probably worth leaving for now, until we limit iconEnd to only a few options.

Comment (ponori) [reply docs/pages/button.js]: 👍

File: docs/pages/button.js [hunk 23]
```diff
@@ -259,245 +212,503 @@ card(
...
+        type="don't"
+        description="Do not add icons to a Button to reinforce the text."
+        defaultCode={`
+          <Button text="Create new Pinterest account" size="lg" color="red" iconEnd='person-add' />
+`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="do"
+        description="Use an IconButton + Tooltip next to the disabled Button if you need to explain why it is disabled,"
```
Comment (dangerismycat) [inline docs/pages/button.js:292]: typo: should end with a period, not a comma

File: docs/pages/button.js [hunk 24]
```diff
@@ -259,245 +212,503 @@ card(
...
+    >
+      <MainSection.Card
+        cardSize="md"
+        defaultCode={`
+function WhiteButtonExample() {
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef();
+  const viewRef = React.useRef();
+
+  React.useEffect(() => {
```
Comment (dangerismycat) [inline docs/pages/button.js:420]: The useEffect and useState here aren't necessary. Just put the Layer+Popover in the markup with no conditional. The reader doesn't need to be able to close the Popover on click.

File: docs/pages/button.js [hunk 25]
```diff
@@ -259,245 +212,503 @@ card(
...
-  <Box padding={2}>
-    <Button text="Default inline button" />
-  </Box>
-  <Box padding={2}>
-    <Button fullWidth text="Full-width button" />
-  </Box>
-</React.Fragment>`}
-  />,
-);
+  React.useEffect(() => {
```
Comment (dangerismycat) [inline docs/pages/button.js:473]: ditto here

File: docs/pages/button.js [hunk 26]
```diff
@@ -259,245 +212,503 @@ card(
...
-  <Example
-    id="ref"
-    name="ref"
-    defaultCode={`
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Ref"
+      description={`
+To control focus, or position any anchor components to IconButton, use ref, as shown in the example below.
```
Comment (dangerismycat) [inline docs/pages/button.js:695]: Run-on sentence, also missing formatting and referencing the wrong component.

"To control focus or position anchored components relative to Button, use `ref` as shown in the example below."

File: docs/pages/button.js [hunk 27]
```diff
@@ -259,245 +212,503 @@ card(
...
+  React.useEffect(() => {
+    setOpen(true)
+  });
+
+  return (
+    <ScrollBoundaryContainer>
+      <Box ref={viewRef} width={300} height={220}>
+        <Box display="flex" justifyContent="center" ref={anchorRef}>
+          <Tabs
+            activeTabIndex={1}
```
Comment (dangerismycat) [inline docs/pages/button.js:429]: Change this to `0` to make this element more obviously a Tab

File: docs/pages/button.js (hunk 14 above)
Comment (dangerismycat) [reply docs/pages/button.js:435]: Agreed with Alberto — see my comment above about an easy fix

File: docs/pages/button.js (hunk 23 above)
Comment (ponori) [reply docs/pages/button.js:292]: Done!

File: docs/pages/button.js (hunk 26 above)
Comment (ponori) [reply docs/pages/button.js:695]: Done!

File: docs/pages/button.js (hunk 27 above)
Comment (ponori) [reply docs/pages/button.js:429]: Done!

File: docs/pages/button.js (hunk 18 above)
Comment (dangerismycat) [reply docs/pages/button.js:585]: @AlbertCarreras The IconButton fix should be in a separate PR: https://github.com/pinterest/gestalt/pull/1691

File: docs/pages/button.js (hunk 24 above)
Comment (ponori) [reply docs/pages/button.js:420]: Strange. When I remove those elements, the Popover doesn't display... Let me keep digging in on this.

Comment (ponori) [reply docs/pages/button.js:420]: Chatted w/ Ryan and we're going to keep in for now as removing this logic prevents the popover from displaying.

File: docs/pages/button.js (hunk 25 above)
Comment (ponori) [reply docs/pages/button.js:473]: Same for this as well.

**Review by dangerismycat — APPROVED** [review]
LGTM!
